sympy/matrices/common.py

```diff
@@ -2218,37 +2218,42 @@ def __neg__(self):
         return self._eval_scalar_mul(-1)
 
     @call_highest_priority('__rpow__')
-    def __pow__(self, num):
+    def __pow__(self, exp):
         if self.rows != self.cols:
             raise NonSquareMatrixError()
         a = self
-        jordan_pow = getattr(a, '_matrix_pow_by_jordan_blocks', None)
-        num = sympify(num)
-        if num.is_Number and num % 1 == 0:
+        exp = sympify(exp)
+        jordan_pow = getattr (a, '_matrix_pow_by_jordan_blocks', None)
+
+        if exp.is_Number and exp % 1 == 0:
             if a.rows == 1:
-                return a._new([[a[0]**num]])
-            if num == 0:
+                return a._new([[a[0]**exp]])
+            if exp == 0:
                 return self._new(self.rows, self.cols, lambda i, j: int(i == j))
```

Comment:
This also can be outside the outer if. It could be if exp.is_zero which might work in more complicated cases. Also can self.eye be used here?

Reply:
Yes I suppose but this is just as valid and quick, again, minimizing changes.

Reply:
Doesn't comparison to zero imply is_zero?

Reply:
Not quite, the converse is true though. The corner cases are obscure but e.g.:
```
In [38]: Integral(x, (x, y, y))                                                                                                   
Out[38]: 
y     
⌠     
⎮ x dx
⌡     
y     

In [39]: Integral(x, (x, y, y)) == 0                                                                                              
Out[39]: False

In [40]: Integral(x, (x, y, y)).is_zero                                                                                           
Out[40]: True
```
This is because == tests structural equality (are these the exact same type of SymPy object) whereas is_zero does basic semantic analysis to check try and establish if an expression is zero or not (it gives True, False or None with None meaning not sure).